File: src/test/java/dev/bothin/testcontainers/java/Test1.java
package dev.bothin.testcontainers.java;

import dev.bothin.testcontainers.TestContainerConfigurationClass;
import dev.bothin.testcontainers.TestContainerLifecycleExtension;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.junit.jupiter.api.Assertions.assertEquals;

@ExtendWith(TestContainerLifecycleExtension.class)
@TestContainerConfigurationClass(ContainerConfiguration.class)
public class Test1 {
    @BeforeEach
    public void setUp() {
        System.out.println("Java Test1 before");
    }

    @Test
    public void when1() {
        System.out.println("Java Test1 when 1");
        System.out.println(System.getProperty("MQTT_HOST"));
        System.out.println(System.getProperty("MQTT_PORT"));
        assertEquals(1, 1);
    }


    @Test
    public void when2() {
        System.out.println("Java Test1 when 2");
        assertEquals(2, 2);
    }
}

File: src/test/kotlin/dev/bothin/testcontainers/Test1.kt
package dev.bothin.testcontainers

import com.github.dockerjava.core.DockerClientBuilder
import io.kotlintest.shouldBe
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.testcontainers.shaded.org.apache.commons.lang.SystemUtils


@ExtendWith(TestContainerLifecycleExtension::class)
@TestContainerConfigurationClass(ContainerConfiguration::class)
class Test1 {

    @BeforeEach
    fun setUp() {
        println("Test1 before")
    }

    @Test
    fun `when 1`() {
        println("Test1 when 1")
        println(System.getProperty("MQTT_HOST"))
        println(System.getProperty("MQTT_PORT"))
        1 shouldBe 1
    }


    @Test
    fun `when 2`() {
        println("Test1 when 2")
        2 shouldBe 2
    }

}